### ISSUE TITLE
Extra Result node on top of CustomScan on PG15

### ISSUE THREAD

**Issue by sb230132** [timescale/timescaledb#4833]
### What type of bug is this?

Incorrect result

### What subsystems and features are affected?

Continuous aggregate

### What happened?

TimescaleDB when compiled against PG15 causes many cagg tests to fail. 
Initial failure is related to permissions which reported error like "permission denied for schema public"
Once permission issues is fixed we get incorrect results as shown below.

```
+++ /Volumes/Work/pg15-regresscheck-testing/build/tsl/test/results/exp_cagg_monthly.out 2022-10-17 09:35:26.000000000 +0530
@@ -1044,6 +1044,7 @@
 (3 rows)

 GRANT USAGE ON FOREIGN SERVER :DATA_NODE_1, :DATA_NODE_2, :DATA_NODE_3 TO PUBLIC;
+GRANT CREATE ON SCHEMA public TO :ROLE_DEFAULT_PERM_USER;
 SET ROLE :ROLE_DEFAULT_PERM_USER;
 CREATE TABLE conditions_dist(
   day DATE NOT NULL,
@@ -1065,7 +1066,7 @@
    MAX(temperature)
 FROM conditions_dist
 GROUP BY bucket;
-NOTICE:  refreshing continuous aggregate "conditions_dist_1m"
+NOTICE:  continuous aggregate "conditions_dist_1m" is already up-to-date
 SELECT mat_hypertable_id AS cagg_id
 FROM _timescaledb_catalog.continuous_agg
 WHERE user_view_name = 'conditions_dist_1m'
@@ -1091,11 +1092,9 @@
 (1 row)

 SELECT * FROM conditions_dist_1m ORDER BY bucket;
-   bucket   | min | max
-------------+-----+-----
- 01-01-2010 | 101 | 131
- 02-01-2010 | 201 | 228
-(2 rows)
+ bucket | min | max
+--------+-----+-----
+(0 rows)
```

### TimescaleDB version affected

2.9.0

### PostgreSQL version used

15.0

### What operating system did you use?

Mac OS X 

### What installation method did you use?

Source

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
Please run below regresscheck tests after fixing permission issues.

cagg_ddl_dist_ht
telemetry_stats
cagg_migrate_integer_dist_ht
cagg_migrate_timestamp_dist_ht
cagg_bgw_dist_ht
cagg_ddl_dist_ht
chunk_api
compression_bgw
data_node
dist_api_calls
dist_commands
dist_compression
dist_copy_format_long
dist_copy_long
dist_ddl
dist_move_chunk
dist_partial_agg
dist_policy
dist_query
exp_cagg_monthly
exp_cagg_origin
exp_cagg_timezone
cagg_concurrent_refresh_dist_ht
cagg_drop_chunks
cagg_multi_dist_ht
continuous_aggs
continuous_aggs_deprecated
```

**Comment by sb230132** [comment]
Fix from PR#4841 has fixed several tests. Tests which are still failing are:
telemetry_stats
dist_commands
dist_ddl
dist_query
dist_partial_agg
continuous_aggs
continuous_aggs_deprecated

**Comment by sb230132** [comment]
This issue is related to #4834
Since on PG15 CustomScan node is wrapped around Result node, we will see change in plans with an extra row in EXPLAIN output.
For ex:
```
 EXPLAIN (COSTS OFF) SELECT * FROM mat_m1;
-                                                                                   QUERY PLAN                                                                                   
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+                                                                                      QUERY PLAN                                                                                      
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  Merge Append
    Sort Key: _materialized_hypertable_59.sum DESC
    ->  Custom Scan (ConstraintAwareAppend)
@@ -2048,13 +2048,14 @@
          Sort Key: (sum(conditions.temperature)) DESC
          ->  HashAggregate
                Group Key: time_bucket('@ 7 days'::interval, conditions.timec)
-               ->  Custom Scan (ChunkAppend) on conditions
-                     Chunks excluded during startup: 1
-                     ->  Index Scan Backward using _hyper_52_111_chunk_conditions_timec_idx on _hyper_52_111_chunk
-                           Index Cond: (timec >= COALESCE(_timescaledb_internal.to_timestamp(_timescaledb_internal.cagg_watermark(59)), '-infinity'::timestamp with time zone))
-                     ->  Index Scan Backward using _hyper_52_125_chunk_conditions_timec_idx on _hyper_52_125_chunk
-                           Index Cond: (timec >= COALESCE(_timescaledb_internal.to_timestamp(_timescaledb_internal.cagg_watermark(59)), '-infinity'::timestamp with time zone))
-(21 rows)
+               ->  Result
+                     ->  Custom Scan (ChunkAppend) on conditions
+                           Chunks excluded during startup: 1
+                           ->  Index Scan Backward using _hyper_52_111_chunk_conditions_timec_idx on _hyper_52_111_chunk
+                                 Index Cond: (timec >= COALESCE(_timescaledb_internal.to_timestamp(_timescaledb_internal.cagg_watermark(59)), '-infinity'::timestamp with time zone))
+                           ->  Index Scan Backward using _hyper_52_125_chunk_conditions_timec_idx on _hyper_52_125_chunk
+                                 Index Cond: (timec >= COALESCE(_timescaledb_internal.to_timestamp(_timescaledb_internal.cagg_watermark(59)), '-infinity'::timestamp with time zone))
+(22 rows)
```
